### ISSUE TITLE
use envoy-build-ubuntu image build error

### ISSUE THREAD

**Issue by headshot289** [envoyproxy/envoy#13796]
### The following is the terminal output
root@99159dbae7b0:/source# ./ci/do_ci.sh bazel.dev
No remote cache bucket is set, skipping setup remote cache.
ENVOY_SRCDIR=/source
start setup_clang.sh
$TEST_TMPDIR defined: output root default is '/build/tmp' and max_idle_secs default is '15'.
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
/source/tools/bazel.rc
Starting local Bazel server and connecting to it...
LLVM_PREFIX = /opt/llvm
BAZELRC_FILE = /source/clang.bazelrc
HEAD is now at 03b4593... remove v1 methods (#108)
building using 4 CPUs
clang toolchain with libc++ configured
bazel fastbuild build with tests...
Building...
Building...
$TEST_TMPDIR defined: output root default is '/build/tmp' and max_idle_secs default is '15'.
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
/source/tools/bazel.rc
INFO: Analyzed target //source/exe:envoy-static (428 packages loaded, 19365 targets configured).
INFO: Found 1 target...
INFO: Deleting stale sandbox base /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/sandbox
INFO: From CcCmakeMakeRule external/envoy/bazel/foreign_cc/zlib/include [for host]:

#### ERROR: /build/tmp/_bazel_bazel/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:177:1: not all outputs were created or valid
Target //source/exe:envoy-static failed to build
INFO: Elapsed time: 52.778s, Critical Path: 17.47s
INFO: 1 process: 1 processwrapper-sandbox.
FAILED: Build did NOT complete successfully

**Comment by yanavlasov** [comment]
Do you have `/build/tmp` directory? Is it writable?

**Comment by headshot289** [comment]
> Do you have `/build/tmp` directory? Is it writable?

Thanks for your reply. 

This directory exists and writable.

root@99159dbae7b0:/build/tmp# pwd
/build/tmp
root@99159dbae7b0:/build/tmp# ls -ll
total 0
drwxr-xr-x 5 root root 160 Oct 26 09:10 _bazel_bazel
drwxr-xr-x 5 root root 160 Oct 28 03:36 _bazel_envoybuild

**Comment by lizan** [comment]
Which image are you using and what version of Envoy are you trying to compile?

**Comment by headshot289** [comment]
It's my problem, resolved after removing the /tmp/envoy-docker-build/tmp/  directory on my host.

**Comment by headshot289** [comment]
> Which image are you using and what version of Envoy are you trying to compile?

@lizan  I used image is envoyproxy/envoy-build-ubuntu tag:b480535e8423b5fd7c102fd30c92f4785519e33a, Envoy on release/v1.16 branch

**Comment by smwyzi** [comment]
i met the same problem, @headshot289 @mattklein123

**Comment by aakatev** [comment]
same error. do you guys have any advice?

**Comment by aakatev** [comment]
Just tried to build it on Linux machine and it works fine. The issue persists on Mac Docker host on all the releases including 1.14-1.16.

**Comment by smwyzi** [comment]
> Just tried to build it on Linux machine and it works fine. The issue persists on Mac Docker host on all the releases including 1.14-1.16.

my docker version is 1.19, here is docker info output:
`
docker info
Client:
 Debug Mode: false
 Plugins:
  scan: Docker Scan (Docker Inc., v0.3.4)

Server:
 Containers: 24
  Running: 0
  Paused: 0
  Stopped: 24
 Images: 67
 Server Version: 19.03.13
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fba4e9a7d01810a393d5d25a3621dc101981175
 runc version: dc9208a3303feef5b3839f4323d9beb36df0a9dd
 init version: fec3683
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.39-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 3.844GiB
 Name: docker-desktop
 ID: T5V5:WBJK:NGDJ:PE62:2FC6:SMYM:6SU3:KJM3:AD6L:B7V6:AVYC:U7G3
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: gateway.docker.internal:3128
 HTTPS Proxy: gateway.docker.internal:3129
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
`

**Comment by smwyzi** [comment]
i have to install virtualbox in my mac, then install ubuntu in virtualbox, then install docker in ubuntu, finally, build envoy in docker in ubuntu in virtualbox in mac ^ω^

**Comment by aakatev** [comment]
> i have to install virtualbox in my mac, then install ubuntu in virtualbox, then install docker in ubuntu, finally, build envoy in docker in ubuntu in virtualbox in mac ^ω^

Interesting.

I wonder if there is some known and undocumented (or missed by me) reason why docker builds on mac hosts are failing.

**Comment by smwyzi** [comment]
> > i have to install virtualbox in my mac, then install ubuntu in virtualbox, then install docker in ubuntu, finally, build envoy in docker in ubuntu in virtualbox in mac ^ω^
> 
> Interesting.
> 
> I wonder if there is some known and undocumented (or missed by me) reason why docker builds on mac hosts are failing.

it seems like that envoy developpers do not use Mac， or this problem can't apeer

**Comment by li-guest** [comment]
Same error in Mac, do you have other solution? @aakatev

**Comment by aakatev** [comment]
@li-guest I gave up trying to build on Mac. The solution is digging into [this script](https://github.com/envoyproxy/envoy/blob/master/ci/do_ci.sh). It was not worth it for me, since the build agents used for Envoy builds are running Linux.

**Comment by boringuy** [comment]
I got the same error when using docker on Mac. I am not familiar with the ci_build script but if it is using shared folder, changing the bazel_cache directory to directory inside the container will solve the problem. I modified my .bazelrc to add --output_user_root=/tmp/bazel_cache startup option.

**Comment by alandtsang** [comment]
Is there a solution?

```
Generating v2 API RST...
Starting local Bazel server and connecting to it...
INFO: Analyzed target @envoy_api_canonical//:v2_protos (300 packages loaded, 1938 targets configured).
INFO: Found 1 target...
INFO: From CcCmakeMakeRule external/envoy/bazel/foreign_cc/zlib/include:

ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:219:21: TreeArtifact external/envoy/bazel/foreign_cc/zlib/include was not created
ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:219:21: TreeArtifact external/envoy/bazel/foreign_cc/copy_zlib/zlib was not created
ERROR: /build/.cache/bazel/_bazel_envoybuild/b570b5ccd0454dc9af9f65ab1833764d/external/envoy/bazel/foreign_cc/BUILD:219:21: not all outputs were created or valid
Aspect //tools/protodoc:protodoc.bzl%protodoc_aspect of @envoy_api_canonical//:v2_protos failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 73.548s, Critical Path: 13.64s
INFO: 1 process: 1 processwrapper-sandbox.
FAILED: Build did NOT complete successfully
```

**Comment by renana** [comment]
I believe the root cause of this is this line: https://github.com/envoyproxy/envoy/blob/master/ci/run_envoy_docker.sh#L55 
It needs to be protected by the `if is_windows` check called above. Changing this solve the build issue for me.

**Comment by jstewmon** [comment]
I ran into this issue too. Having had problems with the gRPC FUSE file sharing feature in the past, I tried switching back to osxfs file sharing, and the problem vanished.

gRPC FUSE became the default in [Docker Desktop Community 2.4.0.0](https://docs.docker.com/docker-for-mac/release-notes/#docker-desktop-community-2400) (released 2020-09-30). If you have a problem with a mounted volume using Docker for Mac, try switching back to osxfs.

![image](https://user-images.githubusercontent.com/909418/107892293-cb2b9b00-6ee9-11eb-88b4-42e8dcbf8773.png)
